### PR TITLE
fix sample to use apollo-runtime and okhttp

### PR DESCRIPTION
**Pull request by brianPlummer**
image

### CODE REVIEW

**Comment by digitalbuddha** [comment]
closes #242

File: apollo-sample/build.gradle [hunk 1]
```diff
@@ -29,8 +29,9 @@ dependencies {
   compile dep.appcompat
   compile dep.rxjava2
   compile dep.rxandroid
-  compile dep.retrofitRx2Adapter
   compile dep.okhttpLoggingInterceptor
+  compile project(path: ':apollo-runtime')
```
Comment (digitalbuddha) [inline apollo-sample/build.gradle]: please depend on the maven artifact here. this way we will know if there was a problem with the published version

Comment (brianPlummer) [inline apollo-sample/build.gradle]: done!

Comment (marwanad) [inline apollo-sample/build.gradle]: might be outside the scope of this PR but should we also add this dependency through the Plugin?

It seems that apollo-runtime would have to be included for all use-cases.

File: apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java [hunk 2]
```diff
@@ -18,25 +21,31 @@
...
     final TextView txtResponse = (TextView) findViewById(R.id.txt_response);
-    SampleApplication application = (SampleApplication) getApplication();
-    application.githuntApiService()
-        .githuntFeed(new FeedQuery(FeedQuery.Variables.builder()
-                .limit(10)
-                .type(FeedType.HOT)
-                .build()))
+    application = (SampleApplication) getApplication();
+
+    Observable<Response<FeedQuery.Data>> query = Observable.fromCallable(new Callable<Response<FeedQuery.Data>>() {
```
Comment (digitalbuddha) [inline apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java]: Wow this looks really ugly.  No fault of yours. Let's try this for now (until I get the Rxjava Call Adapter stuff in).  Move `executeFeedQuery` to `SampleApplication` have that method take a param of limit and return an observable of the Response.  This way the activity can just do application.executeFeedQuery(10).subscribe()

File: apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java [hunk 3]
```diff
@@ -18,25 +21,31 @@
...
+      @Override public Response<FeedQuery.Data> call() throws Exception {
+        return executeFeedQuery();
+      }
+    });
+
+    query
         .subscribeOn(Schedulers.io())
         .observeOn(AndroidSchedulers.mainThread())
         .subscribe(new Observer<Response<FeedQuery.Data>>() {
```
Comment (digitalbuddha) [inline apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java]: please use a `DisposableObserver` which does not have the `OnSubscribe` callback

File: apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java (hunk 2 above)
Comment (digitalbuddha) [inline apollo-sample/src/main/java/com/example/apollographql/sample/MainActivity.java]: Actually theres no need to even do rxjava anymore. there is an apolloclient method .enque that takes a callback and schedules to run in background... use that instead and drop rxjava from sample please

**Comment by digitalbuddha** [comment]
Yes please make issue

File: apollo-sample/build.gradle (hunk 1 above)
Comment (marwanad) [inline apollo-sample/build.gradle]: #132 is useful now.

File: apollo-sample/src/main/java/com/example/apollographql/sample/SampleApplication.java [hunk 4]
```diff
@@ -2,46 +2,35 @@
...
+  private static final String BASE_URL = "https://githunt-api.herokuapp.com/graphql";
+  private ApolloClient apolloClient;
 
   @Override public void onCreate() {
     super.onCreate();
     HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
     loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
-    okHttpClient = new OkHttpClient.Builder()
+    OkHttpClient okHttpClient = new OkHttpClient.Builder()
         .cache(new Cache(new File(getCacheDir(), "okhttp"), 10 * 1024))
```
Comment (digitalbuddha) [inline apollo-sample/src/main/java/com/example/apollographql/sample/SampleApplication.java]: remove cache, it does nothing as all post requests.

**Pull request merged**